src/audio/host-legacy.c

```diff
@@ -226,18 +239,14 @@ void host_common_update(struct host_data *hd, struct comp_dev *dev, uint32_t byt
 {
 	struct comp_buffer *source;
 	struct comp_buffer *sink;
-	int ret;
+	int ret = 0;
 	bool update_mailbox = false;
 	bool send_ipc = false;
 
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
 		source = hd->dma_buffer;
 		sink = hd->local_buffer;
 		ret = dma_buffer_copy_from(source, sink, hd->process, bytes, DUMMY_CHMAP);
-	} else {
-		source = hd->local_buffer;
-		sink = hd->dma_buffer;
-		ret = dma_buffer_copy_to(source, sink, hd->process, bytes, DUMMY_CHMAP);
 	}
```

Comment:
@chihualee any update, this part looks wrong as capture uses the DMA buffer for sink, unless your device has no DMA to copy data to host driver from DSP (i.e. you have shared memory) ?

Reply:
@lgirdwood 

My capture mechanism is almost the same as original legacy-host, 
first data from local buffer to DMA buffer,
then data from DMA buffer to Host buffer.

src/audio/host-legacy.c
Comment on lines -237 to 250
I think this part is doing that capture copying data from local buffer to DMA buffer.
I just move this part to src/audio/host-legacy.c
host_copy_one_shot() around at lines 214
Because it should do first.
![image](https://github.com/user-attachments/assets/f73cb1b7-48fe-4de8-9189-379ea3a57a07)

Reply:
Something unusual is happening here, the directions should not need changed as the code is working for everyone else. Can you check your topology and Linux driver to make sure the topology is setting the buffer and stream direction correctly, I would also try and take a simple "passthrough" pipeline/topology and try this on your system as I dont think we need this change (given its working everywhere else).